node/node_test.go

```diff
@@ -244,6 +245,61 @@ func TestRegisterProtocols_OpenRPC_WS(t *testing.T) {
 	}
 }
 
+func TestDiscoverCompliesOpenRPCSpec(t *testing.T) {
+	stack, err := New(testNodeConfig())
+	if err != nil {
+		t.Fatalf("failed to create protocol stack: %v", err)
+	}
+	defer stack.Close()
+
+	datadir := filepath.Join(os.TempDir(), "node-test")
+	defer os.RemoveAll(datadir)
+
+	stack.config = &DefaultConfig
+	stack.config.HTTPHost = DefaultHTTPHost
+	// enable all available APIs
+	stack.config.HTTPModules = []string{"admin", "debug", "eth", "ethash", "miner", "net", "personal", "rpc", "trace", "txpool", "web3"}
+	stack.config.DataDir = datadir
+
+	if err := stack.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer stack.Close()
+
+	client, err := rpc.Dial("http://localhost:8545")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	res := make(map[string]interface{})
+	err = client.Call(&res, "rpc_discover")
+	if err != nil {
+		t.Errorf("client call: %v", err)
+	}
+
+	openRPCSpecURL := "https://github.com/open-rpc/meta-schema/releases/download/1.14.0/open-rpc-meta-schema.json"
+
+	schemaLoader := gojsonschema.NewSchemaLoader()
+	schemaLoader.Draft = gojsonschema.Draft7 // force schema version
```

Comment:
I enforced using the Draft7 based on the OpenRPC spec, as the schema json doesn't set $schema as appropriate.

Reply:
Noted. If this ever breaks we'll have to investigate more about the JSON Schema they're using served at https://meta.json-schema.tools/.

Reply:
But LGTM.

Reply:
We can leave it on auto, which will use hybrid mode based on [this](https://github.com/xeipuuv/gojsonschema#using-a-specific-draft):

> a hybrid mode is used which merges together functionality of all drafts into one mode